### DISCUSSION TITLE
customergrid or moviegrid

### DISCUSSION THREAD

**Discussion by zprofreelancer** [serenity-is/Serenity#5385] [Help]
Hello,
I'm reading the movie tutorial...but i can't find out how the customergrid.ts or moviegrid.ts automatically create the new customer/movie button and the search button and refresh and column picker buttons... I have also looked at entitygrid but no use.. so, how this is done in code?.. how the customergrid .init() works?.. how to remove the search button for example?

Thanks.

**Comment by minhhungit** [comment]
It is complex to explain, so it is better you should check its code

[Serenity.EntityGrid](https://github.com/serenity-is/Serenity/blob/b8fd791681e91014e5ddc7ee24f3efbeb91f0ca0/src/Serenity.Scripts/CoreLib/UI/DataGrid/EntityGrid.ts)` and [Serenity.EntityDialog](https://github.com/serenity-is/Serenity/blob/b8fd791681e91014e5ddc7ee24f3efbeb91f0ca0/src/Serenity.Scripts/CoreLib/UI/Dialogs/EntityDialog.ts)

That is clue

**Reply by zprofreelancer** [reply]
> 
> 
> It is complex to explain, so it is better you should check its code
> 
> [Serenity.EntityGrid](https://github.com/serenity-is/Serenity/blob/b8fd791681e91014e5ddc7ee24f3efbeb91f0ca0/src/Serenity.Scripts/CoreLib/UI/DataGrid/EntityGrid.ts)` and [Serenity.EntityDialog](https://github.com/serenity-is/Serenity/blob/b8fd791681e91014e5ddc7ee24f3efbeb91f0ca0/src/Serenity.Scripts/CoreLib/UI/Dialogs/EntityDialog.ts)
> 
> That is clue

i don't have entitygrid and entityDialog...May be because i'm using the mvc version.

**Reply by minhhungit** [reply]
![image](https://user-images.githubusercontent.com/2279508/105629090-8a54df00-5e73-11eb-8b7c-6098476c96d3.png)

**Comment by minhhungit** [comment]
Normally to learn Serenity, you will need to read document, **do tutorial**, after that, learn Serenity example, there are a lot examples here http://serenity.is/demo , it will teach you how to work with Grid, Dialog, Editor... it has many cases normally we will use to build application, it is real treasure

Then you can read wiki, wiki also has a lot examples

Then search answer on Github/Issues, 

The last thing is ask here, you also check serenity source code to see how was it implemented.

I trust that after you try above steps you won't ask question like "how this is done in code". Because the code is open, no one hide it, Serenity full code is available and everyone can read it very easy, your work is searching in this repository and check it

Good luck man

**Comment by zprofreelancer** [comment]
the files entitygrid and entityDialog are on the source files before building...i can see that entitygrid.ts contain the definitions of the entitygrid Methods used in customergrid.ts right?

**Comment by minhhungit** [comment]
>the files entitygrid and entityDialog are on the source files before building

Serenity Framework has 2 main repositories on github:
- [Serenity repository](https://github.com/serenity-is/Serenity): contain full source code of Serenity
- [Serene repository](https://github.com/serenity-is/Serene/): Serenity application template, your project is based on this template

The template contains Serenity "dist" package, "dist" package is buit and generated from code in Serenity repository (typescript code)
Remember Serenity has not only typescript code but also it also has c# code

In your project, they just include generated files (dist package, not full typescript code) so you just see javascript file and its typing, you can see in this path : \Scripts\serenity\*.* (ex: Serenity.CoreLib.js) -- **see attachment image**

Finally if you need to check code about any feature of Serenity in your application, the repository you need is https://github.com/serenity-is/Serenity


![image](https://user-images.githubusercontent.com/2279508/105629446-87f38480-5e75-11eb-9e48-b52c6f16b2cf.png)
